tests/t0109_gateway_web_redirects_test.go

```diff
@@ -0,0 +1,391 @@
+		invalidRedirectsDirCID := fixture.MustGetNode("forced").Base32Cid()
+		// INVALID_REDIRECTS_DIR_HOSTNAME="${INVALID_REDIRECTS_DIR_CID}.ipfs.localhost:$GWAY_PORT"
+		invalidDirBaseURL := fmt.Sprintf("%s://%s.ipfs.%s", u.Scheme, invalidRedirectsDirCID, u.Host)
+		// TOO_LARGE_REDIRECTS_DIR_CID=$(ipfs resolve -r /ipfs/$CAR_ROOT_CID/too-large | cut -d "/" -f3)
+		tooLargeRedirectsDirCID := fixture.MustGetNode("too-large").Base32Cid()
+		// TOO_LARGE_REDIRECTS_DIR_HOSTNAME="${TOO_LARGE_REDIRECTS_DIR_CID}.ipfs.localhost:$GWAY_PORT"
+		tooLargeDirBaseURL := fmt.Sprintf("%s://%s.ipfs.%s", u.Scheme, tooLargeRedirectsDirCID, u.Host)
+
+		tests = append(tests, SugarTests{
+			// # if accessing a path that doesn't exist, read _redirects and fail parsing, and return error
+			// test_expect_success "invalid file: request for $INVALID_REDIRECTS_DIR_HOSTNAME/not-found returns error about invalid redirects file" '
+			//   curl -sD - --resolve $INVALID_REDIRECTS_DIR_HOSTNAME:127.0.0.1 "http://$INVALID_REDIRECTS_DIR_HOSTNAME/not-found" > response &&
+			//   test_should_contain "500" response &&
+			//   test_should_contain "could not parse _redirects:" response &&
+			//   test_should_contain "forced redirects (or \"shadowing\") are not supported" response
+			// '
+			{
+				Name: "invalid file: request for $INVALID_REDIRECTS_DIR_HOSTNAME/not-found returns error about invalid redirects file",
+				Hint: `if accessing a path that doesn't exist, read _redirects and fail parsing, and return error`,
+				Request: Request().
+					URL("%s/not-found", invalidDirBaseURL),
+				Response: Expect().
+					Status(500).
+					Body(
+						And(
+							Contains("could not parse _redirects:"),
+							Contains("forced redirects (or \"shadowing\") are not supported"),
+						),
+					),
+			},
+			// # if accessing a path that doesn't exist and _redirects file is too large, return error
+			// test_expect_success "invalid file: request for $TOO_LARGE_REDIRECTS_DIR_HOSTNAME/not-found returns error about too large redirects file" '
+			//   curl -sD - --resolve $TOO_LARGE_REDIRECTS_DIR_HOSTNAME:127.0.0.1 "http://$TOO_LARGE_REDIRECTS_DIR_HOSTNAME/not-found" > response &&
+			//   test_should_contain "500" response &&
+			//   test_should_contain "could not parse _redirects:" response &&
+			//   test_should_contain "redirects file size cannot exceed" response
+			// '
+			{
+				Name: "invalid file: request for $TOO_LARGE_REDIRECTS_DIR_HOSTNAME/not-found returns error about too large redirects file",
+				Hint: `if accessing a path that doesn't exist and _redirects file is too large, return error`,
+				Request: Request().
+					URL("%s/not-found", tooLargeDirBaseURL),
+				Response: Expect().
+					Status(500).
+					Body(
+						And(
+							Contains("could not parse _redirects:"),
+							Contains("redirects file size cannot exceed"),
+						),
+					),
+			},
+		}...)
+	}
+
+	if specs.SubdomainGateway.IsEnabled() {
+		Run(t, unwrapTests(t, tests).Build())
```

Comment:
@galargh I came up with a new pattern for subdomain:
instead of explicitly calling a function on every test:
https://github.com/ipfs/gateway-conformance/blob/023888a768b3d9d639fec2ed5e898d52de92b573/tests/t0114_gateway_subdomains_test.go#L49-L56

`unwrapTests` takes a bunch of tests relying on a subdomain, and unwrap them, it's much more natural when you write the tests.

Not perfect, but a step in a better direction I think, maybe that'll some ideas on your side :)